webhooks/virtualmachine/mutation/virtualmachine_mutator.go

```diff
@@ -216,22 +222,51 @@ func SetNextRestartTime(
 		`may only be set to "now"`)
 }
 
-// AddDefaultNetworkInterface adds default network interface to a VM if the NoNetwork annotation is not set
-// and no NetworkInterface is specified.
-// Return true if default NetworkInterface is added, otherwise return false.
-func AddDefaultNetworkInterface(ctx *pkgctx.WebhookRequestContext, client ctrlclient.Client, vm *vmopv1.VirtualMachine) bool {
-	// Continue to support this ad-hoc v1a1 annotation. I don't think need or want to have this annotation
-	// in v1a2: Disabled mostly already covers it. We could map between the two for version conversion, but
-	// they do mean slightly different things, and kind of complicated to know what to do like if the annotation
-	// is removed.
-	if _, ok := vm.Annotations[v1alpha1.NoDefaultNicAnnotation]; ok {
-		return false
-	}
+func setDefaultNetworkInterfaceNetwork(
+	vm *vmopv1.VirtualMachine,
+	ifaceIdx int,
+	networkRef common.PartialObjectRef,
+) bool {
+	ifaceNetwork := vm.Spec.Network.Interfaces[ifaceIdx].Network
+	var updated bool
```

Comment:
nit: looks you could drop this var and just `return true` instead of `updated = true`. And with those early returns you can drop the `else {` below